### PR TITLE
CRIMRE-83 Search by application id.

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change
Add search endpoint and allow filtering by application_id.

This feature is part of the initial spike of the search endpoint for Review.

## Link to relevant ticket
[CRIMRE-44](https://dsdmoj.atlassian.net/browse/CRIMRE-44)

## Notes for reviewer / how to test

Filtering by application_id will be used on review to scope searched applications to those of a given caseworker.

### CODE REVIEW

**Review by arthurashman — APPROVED** [review]
LGTM

File: app/api/datastore/base.rb [hunk 1]
```diff
@@ -23,16 +23,16 @@ class Base < Grape::API
...
         )
 
         optional(
           :per_page,
           type: Integer,
-          default: 20,
+          default: Pagination::DEFAULT_PER_PAGE,
           desc: 'Number of results to return per page.',
-          values: 1..200
+          values: 1..Pagination::MAX_PER_PAGE
```
Comment (zheileman) [inline app/api/datastore/base.rb:35]: On Apply we've been thinking if allowing to pass 0 `per_page` might be a sensible approach for requests where we are only interested in the total number of records. That will essentially return to us the paginator object with the metadata, but an empty records collection.

Not a big prob, as right now we are using limit 1 which is still efficient.
This is not a blocker, just crossed my mind.

File: app/api/datastore/base.rb [hunk 2]
```diff
@@ -23,16 +23,16 @@ class Base < Grape::API
         optional(
           :page,
           type: Integer,
-          default: 1,
+          default: Pagination::DEFAULT_PAGE,
```
Comment (willmcb) [inline app/api/datastore/base.rb:26]: Nice to remove those magic numbers 👍

File: spec/api/datastore/v2/searches/filter_by_application_id_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe 'searches' do
```
Comment (willmcb) [inline spec/api/datastore/v2/searches/filter_by_application_id_spec.rb:3]: Should we add a spec for the ordering as well? i.e. `:submitted_at`?

File: app/services/operations/search.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+module Operations
+  class Search
+    def initialize(search:, pagination:)
+      @filter = SearchFilter.new(search)
+      @pagination = Pagination.new(pagination)
+      @scope = CrimeApplication
+    end
+
+    def call
+      query
```
Comment (willmcb) [inline app/services/operations/search.rb:10]: How is query i.e. `CrimeApplication` able to recieve `page` here? Is it due to the presenters on L19 in `app/api/datastore/v2/searches.rb` little confused by this.

File: app/api/datastore/base.rb (hunk 1 above)
Comment (timpeat) [reply app/api/datastore/base.rb:35]: Good point. We can always add them to the headers and do an headers only request.

File: app/services/operations/search.rb (hunk 4 above)
Comment (timpeat) [reply app/services/operations/search.rb:10]: I'll try to make this clearer in the next pr.

**Pull request merged**